File: system-tests/double-start-agent/src/test/java/com/teamscale/client/DoubleStartAgentTest.java
package com.teamscale.client;

import org.junit.jupiter.api.Test;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;
import java.util.List;
import java.util.stream.Stream;

import static org.assertj.core.api.Assertions.assertThat;

public class DoubleStartAgentTest {

	private static final Path LOG_DIRECTORY = Paths.get("logTest").resolve("logs");

	@Test
	public void systemTest() throws IOException {
		assertThat(LOG_DIRECTORY).exists();
		List<String> lines = Files.readAllLines(LOG_DIRECTORY.resolve("teamscale-jacoco-agent.log"));
		Stream<String> agentStartLines = lines.stream().filter(line -> line.contains("Starting JaCoCo agent"));
		assertThat(agentStartLines).hasSize(1);
	}
}

File: system-tests/double-start-agent/src/test/kotlin/com/teamscale/client/DoubleStartAgentTest.kt
package com.teamscale.client

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import java.io.IOException
import java.nio.file.Paths
import kotlin.io.path.readLines

class DoubleStartAgentTest {
	@Test
	@Throws(IOException::class)
	fun systemTest() {
		assertThat(LOG_DIRECTORY).exists()
		val lines = LOG_DIRECTORY.resolve("teamscale-jacoco-agent.log").readLines()
		val agentStartLines = lines.filter { it.contains("Starting Teamscale Java Profiler") }
		assertThat(agentStartLines).hasSize(1)
	}

	companion object {
		private val LOG_DIRECTORY = Paths.get("logTest").resolve("logs")
	}
}
